server/controllers/api.py

```diff
@@ -351,14 +352,24 @@ def store_backup(self, user):
         if not assignment:
             raise ValueError('Assignment does not exist')
         lock_flag = not assignment['active']
+        past_due = dt.utcnow() > assignment['due_date']
 
         # Do not allow submissions after the lock date
-        elgible_submit = args['submit'] and not lock_flag
+        eligible_submit = args['submit'] and not lock_flag
         backup = make_backup(user, assignment['id'], args['messages'],
-                             elgible_submit)
-        if args['submit'] and lock_flag:
-            raise ValueError('Late Submission of {}'.format(args['assignment']))
-        if elgible_submit and assignment['autograding_key']:
+                             eligible_submit)
+        if args['submit'] and past_due:
+            assign_obj = models.Assignment.by_name(args['assignment'])
+            extension = models.Extension.get_extension(user, assign_obj)
+            # Submissions after the deadline with an extension are allowed
```

Comment:
It's more that the custom time is set - if the lock date hasn't passed - we'll still accept it regardless of extensions.